smithy4s-client-producer-tests/src/it/scala/kinesis4cats/smithy4s/client/producer/KinesisProducerSpec.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2023-2023 etspaceman
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kinesis4cats.smithy4s.client.producer
+
+import java.util.UUID
+
+import cats.effect._
+import cats.effect.syntax.all._
+import com.amazonaws.kinesis.PutRecordsInput
+import com.amazonaws.kinesis.PutRecordsOutput
+import org.http4s.ember.client.EmberClientBuilder
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+import smithy4s.aws.kernel.AwsRegion
+
+import kinesis4cats.kcl.CommittableRecord
+import kinesis4cats.kcl.localstack.LocalstackKCLConsumer
+import kinesis4cats.kcl.logging.instances.show._
+import kinesis4cats.logging.instances.show._
+import kinesis4cats.producer.Producer
+import kinesis4cats.producer.ProducerSpec
+import kinesis4cats.producer.logging.instances.show._
+import kinesis4cats.smithy4s.client.localstack.LocalstackKinesisClient
+import kinesis4cats.smithy4s.client.logging.instances.show._
+import kinesis4cats.smithy4s.client.producer.localstack.LocalstackKinesisProducer
+import kinesis4cats.syntax.bytebuffer._
+
+class KinesisProducerSpec
+    extends ProducerSpec[
+      PutRecordsInput,
+      PutRecordsOutput,
+      CommittableRecord[IO]
+    ]() {
+  override lazy val streamName: String =
+    s"kinesis-client-producer-spec-${UUID.randomUUID().toString()}"
+
+  def emberClientResource = EmberClientBuilder
```

Comment:
This is the test that can be run to reproduce it. You can do the following in an SBT shell

```
;IT/dockerComposeUp;smithy4s-client-producer-tests/IT/test
```